src/lib.rs

```diff
@@ -118,6 +132,12 @@ impl<T: FileLoader> VfsInternal<T> {
         }
         result
     }
+
+    fn get_line(&self, path: &Path, line: usize) -> Option<String> {
+        self.load_file(path);
+        let files = self.files.lock().unwrap();
+        files.get(path).map(|f| f.get_line(line).to_string())
```

Comment:
this is slightly odd semantics in that if the the file is not present (which should be impossible, and currently is because of the `load_file` call) then we can return None, but if the line is not in the file (which shouldn't happen, but is only ensured by the client, then `File::get_line` will panic). Probably the easiest thing to do is to panic if the file is not in the map, and punt on doing it properly. Alternatively, you could change File::get_line to return an Option and change this `map` to `and_then`.